cmd/k8s-kata-manager/main.go

```diff
@@ -331,7 +344,24 @@ func (w *worker) Run(c *cli.Context) error {
 	return nil
 }
 
-// CleanUp reverts the containerd config to remove the nvidia-container-runtime
+func (w *worker) getRuntimeConfig() (runtime.Runtime, error) {
+	var ctrdConfig runtime.Runtime
+	var err error
+	if w.Runtime == api.CRIO.String() {
+		options := runtime.Options{Path: w.CrioConfig, RuntimeType: "vm", PodAnnotations: []string{"io.katacontainers.*"}}
+		ctrdConfig, err = crio.Setup(&options)
+	} else if w.Runtime == api.Containerd.String() {
+		options := runtime.Options{Path: w.ContainerdConfig, RuntimeType: "io.containerd.kata.v2", PodAnnotations: []string{"io.katacontainers.*"}, Socket: w.ContainerdSocket}
+		ctrdConfig, err = containerd.Setup(&options)
+	}
+	if err != nil {
+		klog.Errorf("error creating containerd.config client : %s", err)
+		return nil, err
+	}
+	return ctrdConfig, nil
+}
+
+// CleanUp reverts the runtime config added by kata manager
```

Comment:
Question -- shouldn't we update `CleanUp` to be agnostic to the runtime in use? Right now it is hardcoded for containerd.

Reply:
Updated to be generic

Reply:
Also update the logs in this function to not be specific to containerd.